common/OpTestSOL.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python2
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2018
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+'''
+Multithreaded library
+---------------------
+This adds a new multithreaded library with having different
+variants of thread based SSH/SOL session runs, each thread logs
+to a different log file.
+'''
+
+import random
+import unittest
+import time
+import threading
+import pexpect
+
+import OpTestConfiguration
+from OpTestSystem import OpSystemState
+from Exceptions import CommandFailed
+from OpTestIPMI import IPMIConsoleState
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class OpSOLMonitorThread(threading.Thread):
+    '''
+    This thread just monitors the SOL console for any failures when tests are running
+    on other SSH threads
+    '''
+    def __init__(self, threadID, name, execution_time=None):
+        threading.Thread.__init__(self)
+        self.threadID = threadID
+        self.name = name
+        self.execution_time = execution_time
+        conf = OpTestConfiguration.conf
+        self.system = conf.system()
+        self.system.goto_state(OpSystemState.OS)
+        self.c = self.system.console.get_console()
+        self.c_terminate = False;
+
+
+    def run(self):
+        log.debug("Starting %s" % self.name)
+        if self.execution_time:
+            self.timeout_run()
+        else:
+            self.nontimeout_run()
+        log.debug("Exiting %s" % self.name)
+
+    def timeout_run(self):
+        execution_time = time.time() + 60 * self.execution_time
+        log.debug("Starting %s for new SOL thread" % self.name)
+        while True:
+            try:
+                self.c.expect("\n", timeout=60)
+            except pexpect.TIMEOUT:
+                pass
+            except pexpect.EOF:
+                self.c.state = IPMIConsoleState.DISCONNECTED
```

Comment:
Suggest to use self.c.close which does the proper clear_state which is necessary for allowing the automatic prompt setup (PS1_set) to be caught and properly reinitialized.